<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Data_processing â€¢ SelectSim</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Data_processing">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SelectSim</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1.5</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/data_processing.html">Data_processing</a></li>
    <li><a class="dropdown-item" href="../articles/introduction.html">Introduction to finding evolutionary dependencies in cancer data using SelectSim algorithm</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Data_processing</h1>
                        <h4 data-toc-skip class="author">Arvind
Iyer</h4>
            
            <h4 data-toc-skip class="date">2025-04-04</h4>
      

      <div class="d-none name"><code>data_processing.Rmd</code></div>
    </div>

    
    
<ul>
<li>The goal of SelectSim pacakge is to implement the SelectSim
methodology to infer inter-dependencies between functional alterations
in cancer.</li>
<li>
<code>functional</code> package provides function to generate the
backgorund model and other utilites functions.</li>
</ul>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<ul>
<li>You can install the development version of SelectSim from <a href="https://github.com/CSOgroup/SelectSim" class="external-link">GitHub</a> with:</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"CSOgroup/SelectSim"</span>,dependencies <span class="op">=</span> <span class="cn">TRUE</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<ul>
<li>For more details on installation refer to <a href="https://github.com/CSOgroup/SelectSim/blob/main/INSTALLATION.md" class="external-link">INSTALLATION</a>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h3>
<ul>
<li><p>We will process LUAD MAF dataset from TCGA provided with the
package.</p></li>
<li><p>We will also use oncokb v3.9 cancer genes and mutations to filter
the maf to create the gam provided with the pacakge.</p></li>
<li><p>NOTE: This an example to prcoess the MAF file. To know more about
how we prcossed the all MAF file to run_data object, please refer to
SelectSim_analysis <a href="https://github.com/CSOgroup/SelectSim_analysis" class="external-link">repository</a>.</p></li>
</ul>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://csogroup.github.io/SelectSim/">SelectSim</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jabiru/tictoc" class="external-link">tictoc</a></span><span class="op">)</span></span>
<span><span class="co">## Load the data provided with the package</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">luad_maf</span>, package <span class="op">=</span> <span class="st">"SelectSim"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">oncokb_genes</span>, package <span class="op">=</span> <span class="st">"SelectSim"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">oncokb_truncating_genes</span>, package <span class="op">=</span> <span class="st">"SelectSim"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">variant_catalogue</span>, package <span class="op">=</span> <span class="st">"SelectSim"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check the MAF</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">luad_maf</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 220734      8</span></span></code></pre></div>
<ul>
<li>Let print number of lines and number of samples</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input_maf</span> <span class="op">&lt;-</span> <span class="va">luad_maf</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">'##### Number of lines ####'</span>,<span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">input_maf</span><span class="op">)</span>,sep<span class="op">=</span><span class="st">"-&gt;"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "##### Number of lines ####-&gt;220734"</span></span>
<span><span class="va">genes_to_consider</span> <span class="op">=</span>  <span class="va">oncokb_genes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">'##### Number of genes ####'</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">genes_to_consider</span><span class="op">)</span>,sep<span class="op">=</span><span class="st">"-&gt;"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "##### Number of genes ####-&gt;396"</span></span></code></pre></div>
<ul>
<li>Let create a table schema of mutations to conisder and columns
defined in maf file</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mutation_type</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>      <span class="st">'ignore'</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Silent"</span>,<span class="st">"Intron"</span>,<span class="st">"RNA"</span>,<span class="st">"3'UTR"</span>,<span class="st">"5'UTR"</span>,<span class="st">"5'Flank"</span>,<span class="st">"3'Flank"</span>,<span class="st">"IGR"</span><span class="op">)</span>,</span>
<span>      <span class="st">'truncating'</span><span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Frame_Shift_Del'</span>,<span class="st">'Frame_Shift_Ins'</span>,<span class="st">'In_Frame_Del'</span>,<span class="st">'In_Frame_Ins'</span>,<span class="st">'Nonsense_Mutation'</span>,<span class="st">'Nonstop_Mutation'</span>,<span class="st">'Splice_Region'</span>,<span class="st">'Splice_Site'</span>,<span class="st">'Translation_Start_Site'</span><span class="op">)</span>,</span>
<span>      <span class="st">'missense'</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Missense_Mutation'</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">custom_maf_schema</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    <span class="st">'name'</span> <span class="op">=</span> <span class="st">'custom_maf'</span>,</span>
<span>    <span class="st">'column'</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>          <span class="st">'gene'</span> <span class="op">=</span> <span class="st">'Hugo_Symbol'</span></span>
<span>        , <span class="st">'gene.name'</span> <span class="op">=</span> <span class="st">'Hugo_Symbol'</span></span>
<span>        , <span class="st">'sample'</span> <span class="op">=</span> <span class="st">'sample'</span></span>
<span>        , <span class="st">'sample.name'</span> <span class="op">=</span> <span class="st">'sample'</span></span>
<span>        , <span class="st">'mutation.type'</span> <span class="op">=</span> <span class="st">'Variant_Classification'</span></span>
<span>        , <span class="st">'mutation'</span> <span class="op">=</span> <span class="st">'HGVSp_Short'</span></span>
<span>        <span class="op">)</span>,</span>
<span>        <span class="st">'mutation.type'</span> <span class="op">=</span> <span class="va">mutation_type</span></span>
<span><span class="op">)</span></span></code></pre></div>
<ul>
<li>Number of samples in the maf file</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mut_samples</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">input_maf</span><span class="op">[</span>, <span class="va">custom_maf_schema</span><span class="op">$</span><span class="va">column</span><span class="op">$</span><span class="va">sample</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">'##### Number of samples ####'</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">mut_samples</span><span class="op">)</span>,sep<span class="op">=</span><span class="st">"-&gt;"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "##### Number of samples ####-&gt;502"</span></span></code></pre></div>
<ul>
<li>Filter the maf file to include oncokb cancer genes</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">maf_genes</span> <span class="op">=</span> <span class="fu"><a href="../reference/filter_maf_gene.name.html">filter_maf_gene.name</a></span><span class="op">(</span><span class="va">input_maf</span>, genes <span class="op">=</span> <span class="va">genes_to_consider</span>, gene.col <span class="op">=</span> <span class="va">custom_maf_schema</span><span class="op">$</span><span class="va">column</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">'##### Number of lines ####'</span>,<span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">maf_genes</span><span class="op">)</span>,sep<span class="op">=</span><span class="st">"-&gt;"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "##### Number of lines ####-&gt;6708"</span></span></code></pre></div>
<div class="section level4">
<h4 id="genrating-the-gams">Genrating the GAMs<a class="anchor" aria-label="anchor" href="#genrating-the-gams"></a>
</h4>
<ul>
<li>Let generate the truncating data
<ul>
<li>We generate the GAM consider the genes to consider truncating
mutations.</li>
<li>We also create a TMB dataframe which takes all truncating mutation
of all genes in consideration.</li>
</ul>
</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Creating Truncating GAM</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/tictoc/man/tic.html" class="external-link">tic</a></span><span class="op">(</span><span class="st">'##### Creating Truncating GAM ####'</span><span class="op">)</span></span>
<span>    <span class="va">maf_trunc</span> <span class="op">=</span> <span class="fu"><a href="../reference/filter_maf_truncating.html">filter_maf_truncating</a></span><span class="op">(</span><span class="va">maf_genes</span>,genes<span class="op">=</span><span class="va">oncokb_truncating_genes</span>, <span class="va">custom_maf_schema</span><span class="op">)</span></span>
<span>    <span class="va">input_maf_trunc</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/filter_maf_truncating.html">filter_maf_truncating</a></span><span class="op">(</span><span class="va">input_maf</span>, <span class="va">custom_maf_schema</span><span class="op">)</span></span>
<span>    <span class="va">truncating_tmb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="st">'sample'</span><span class="op">=</span><span class="va">mut_samples</span>,<span class="st">'mutation'</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">mut_samples</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">truncating_tmb</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">mut_samples</span></span>
<span>    <span class="va">temp</span> <span class="op">&lt;-</span> <span class="va">input_maf_trunc</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">temp</span><span class="op">$</span><span class="va">sample</span></span>
<span>    <span class="va">truncating_tmb</span><span class="op">[</span><span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">truncating_tmb</span><span class="op">$</span><span class="va">sample</span>,<span class="va">temp</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span>,<span class="op">]</span><span class="op">$</span><span class="va">mutation</span> <span class="op">&lt;-</span><span class="va">temp</span><span class="op">[</span><span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">truncating_tmb</span><span class="op">$</span><span class="va">sample</span>,<span class="va">temp</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span>,<span class="st">'n'</span><span class="op">]</span></span>
<span>    <span class="va">tcga_truc_gam</span> <span class="op">=</span> <span class="fu"><a href="../reference/maf2gam.html">maf2gam</a></span><span class="op">(</span><span class="va">maf_trunc</span>,</span>
<span>                     sample.col <span class="op">=</span> <span class="va">custom_maf_schema</span><span class="op">$</span><span class="va">column</span><span class="op">$</span><span class="va">sample</span>,</span>
<span>                     gene.col <span class="op">=</span> <span class="va">custom_maf_schema</span><span class="op">$</span><span class="va">column</span><span class="op">$</span><span class="va">gene</span>,</span>
<span>                     value.var <span class="op">=</span> <span class="st">'Variant_Classification'</span>,</span>
<span>                     samples <span class="op">=</span> <span class="va">mut_samples</span>,</span>
<span>                     genes <span class="op">=</span> <span class="va">genes_to_consider</span>,</span>
<span>                     fun.aggregate <span class="op">=</span> <span class="va">length</span>,</span>
<span>                     binarize<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                     fill<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span>
<span>    <span class="va">truncating_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'gam'</span><span class="op">=</span><span class="va">tcga_truc_gam</span>,</span>
<span>                            <span class="st">'tmb'</span><span class="op">=</span><span class="va">truncating_tmb</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/tictoc/man/tic.html" class="external-link">toc</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; ##### Creating Truncating GAM ####: 0.228 sec elapsed</span></span></code></pre></div>
<ul>
<li>Let generate the Missense data
<ul>
<li>We generate the GAM with genes and hotspot mutations from oncokb
v3.9.</li>
<li>We also create a TMB dataframe which takes all Missense mutation of
all genes in consideration.</li>
</ul>
</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Creating Missense GAM</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/tictoc/man/tic.html" class="external-link">tic</a></span><span class="op">(</span><span class="st">'##### Creating Missense GAM ####'</span><span class="op">)</span></span>
<span>    <span class="va">maf_valid</span> <span class="op">=</span> <span class="fu"><a href="../reference/filter_maf_schema.html">filter_maf_schema</a></span><span class="op">(</span><span class="va">input_maf</span>,</span>
<span>                             schema <span class="op">=</span> <span class="va">custom_maf_schema</span>,</span>
<span>                             column <span class="op">=</span> <span class="st">'mutation.type'</span>,</span>
<span>                             values <span class="op">=</span> <span class="va">custom_maf_schema</span><span class="op">[[</span><span class="st">'mutation.type'</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">'ignore'</span><span class="op">]</span><span class="op">]</span>,</span>
<span>                             inclusive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>    <span class="va">missense_maf</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/filter_maf_mutation.type.html">filter_maf_mutation.type</a></span><span class="op">(</span><span class="va">input_maf</span>,</span>
<span>                                      variants <span class="op">=</span> <span class="st">'Missense_Mutation'</span>,</span>
<span>                                      variant.col <span class="op">=</span> <span class="va">custom_maf_schema</span><span class="op">$</span><span class="va">column</span><span class="op">$</span><span class="va">mutation.type</span><span class="op">)</span></span>
<span>    <span class="va">missense_tmb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="st">'sample'</span><span class="op">=</span><span class="va">mut_samples</span>,<span class="st">'mutation'</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">mut_samples</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">missense_tmb</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">mut_samples</span></span>
<span>    <span class="va">temp</span> <span class="op">&lt;-</span> <span class="va">missense_maf</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">temp</span><span class="op">$</span><span class="va">sample</span></span>
<span>    <span class="va">missense_tmb</span><span class="op">[</span><span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">missense_tmb</span><span class="op">$</span><span class="va">sample</span>,<span class="va">temp</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span>,<span class="op">]</span><span class="op">$</span><span class="va">mutation</span> <span class="op">&lt;-</span><span class="va">temp</span><span class="op">[</span><span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">missense_tmb</span><span class="op">$</span><span class="va">sample</span>,<span class="va">temp</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span>,<span class="st">'n'</span><span class="op">]</span></span>
<span>    <span class="va">t_m</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="va">maf_valid</span><span class="op">[[</span><span class="va">custom_maf_schema</span><span class="op">$</span><span class="va">column</span><span class="op">$</span><span class="va">mutation</span><span class="op">]</span><span class="op">]</span>,<span class="fl">3</span>,<span class="fl">1000</span><span class="op">)</span></span>
<span>    <span class="va">t_m1</span> <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">'[A-Z]*$'</span>, <span class="st">''</span>, <span class="va">t_m</span><span class="op">)</span></span>
<span>    <span class="va">maf_valid</span><span class="op">$</span><span class="va">HGVSp_Short_fixed</span> <span class="op">=</span> <span class="va">t_m1</span></span>
<span>    <span class="va">maf_hotspot</span> <span class="op">=</span> <span class="fu"><a href="../reference/filter_maf_mutations.html">filter_maf_mutations</a></span><span class="op">(</span><span class="va">maf_valid</span>,</span>
<span>                                  <span class="va">variant_catalogue</span>,</span>
<span>                                  maf.col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">custom_maf_schema</span><span class="op">$</span><span class="va">column</span><span class="op">$</span><span class="va">gene</span>, <span class="st">'HGVSp_Short_fixed'</span><span class="op">)</span>,</span>
<span>                                  values.col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'gene'</span>, <span class="st">'mut'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>    <span class="va">missense_tcga_gam</span> <span class="op">=</span> <span class="fu"><a href="../reference/maf2gam.html">maf2gam</a></span><span class="op">(</span><span class="va">maf_hotspot</span>,</span>
<span>                     sample.col <span class="op">=</span> <span class="va">custom_maf_schema</span><span class="op">$</span><span class="va">column</span><span class="op">$</span><span class="va">sample</span>,</span>
<span>                     gene.col <span class="op">=</span> <span class="va">custom_maf_schema</span><span class="op">$</span><span class="va">column</span><span class="op">$</span><span class="va">gene</span>,</span>
<span>                     value.var <span class="op">=</span> <span class="st">'Variant_Classification'</span>,</span>
<span>                     samples <span class="op">=</span> <span class="va">mut_samples</span>,</span>
<span>                     genes <span class="op">=</span> <span class="va">genes_to_consider</span>,</span>
<span>                     fun.aggregate <span class="op">=</span> <span class="va">length</span>,</span>
<span>                     binarize<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                     fill<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span>
<span>    <span class="va">missesne_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'gam'</span><span class="op">=</span><span class="va">missense_tcga_gam</span>,</span>
<span>                          <span class="st">'tmb'</span><span class="op">=</span><span class="va">missense_tmb</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/tictoc/man/tic.html" class="external-link">toc</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; ##### Creating Missense GAM ####: 1.873 sec elapsed</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="genrating-the-run_object-to-run-selectx">Genrating the run_object to run SelectX<a class="anchor" aria-label="anchor" href="#genrating-the-run_object-to-run-selectx"></a>
</h4>
<ul>
<li>We create a run_object data which is list object which consists of
<ul>
<li>M: a list object of GAMs which is presence absence matrix of
alterations</li>
<li>tmb: a list object of tumor mutation burden as data frame with
column names (should be) as sample and mutationn</li>
<li>sample.class a named vector of sample annotations</li>
<li>alteration.class a named vector of alteration annotations</li>
</ul>
</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_to_take</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">missesne_data</span><span class="op">$</span><span class="va">gam</span><span class="op">)</span></span>
<span><span class="va">order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">missesne_data</span><span class="op">$</span><span class="va">gam</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'M'</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'missense'</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">missesne_data</span><span class="op">$</span><span class="va">gam</span><span class="op">[</span><span class="va">order</span>,<span class="va">gene_to_take</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                     <span class="st">'truncating'</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">truncating_data</span><span class="op">$</span><span class="va">gam</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">missesne_data</span><span class="op">$</span><span class="va">gam</span><span class="op">[</span><span class="va">order</span>,<span class="op">]</span><span class="op">)</span>,<span class="va">gene_to_take</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            <span class="st">'tmb'</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'missense'</span><span class="op">=</span><span class="va">missesne_data</span><span class="op">$</span><span class="va">tmb</span><span class="op">[</span><span class="va">order</span>,<span class="op">]</span>,</span>
<span>                       <span class="st">'truncating'</span><span class="op">=</span><span class="va">truncating_data</span><span class="op">$</span><span class="va">tmb</span><span class="op">[</span><span class="va">order</span>,<span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">alteration_covariates</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">'MUT'</span>,<span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">missesne_data</span><span class="op">$</span><span class="va">gam</span><span class="op">[</span><span class="va">order</span>,<span class="va">gene_to_take</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">alteration_covariates</span><span class="op">)</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">missesne_data</span><span class="op">$</span><span class="va">gam</span><span class="op">[</span><span class="va">order</span>,<span class="va">gene_to_take</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">sample_covariates</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">'LUAD'</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">order</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sample_covariates</span><span class="op">)</span><span class="op">&lt;-</span><span class="va">order</span></span>
<span><span class="va">run_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'M'</span><span class="op">=</span><span class="va">data</span>,<span class="st">'sample.class'</span> <span class="op">=</span> <span class="va">sample_covariates</span>,<span class="st">'alteration.class'</span> <span class="op">=</span> <span class="va">alteration_covariates</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">run_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 3</span></span>
<span><span class="co">#&gt;  $ M               :List of 2</span></span>
<span><span class="co">#&gt;   ..$ M  :List of 2</span></span>
<span><span class="co">#&gt;   .. ..$ missense  : num [1:396, 1:502] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;   .. .. ..- attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">#&gt;   .. .. .. ..$ : chr [1:396] "AKT1" "ATM" "BRAF" "CDKN2A" ...</span></span>
<span><span class="co">#&gt;   .. .. .. ..$ : chr [1:502] "TCGA-05-4244-01" "TCGA-05-4249-01" "TCGA-05-4250-01" "TCGA-05-4382-01" ...</span></span>
<span><span class="co">#&gt;   .. ..$ truncating: num [1:396, 1:502] 0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;   .. .. ..- attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">#&gt;   .. .. .. ..$ : chr [1:396] "AKT1" "ATM" "BRAF" "CDKN2A" ...</span></span>
<span><span class="co">#&gt;   .. .. .. ..$ : chr [1:502] "TCGA-05-4244-01" "TCGA-05-4249-01" "TCGA-05-4250-01" "TCGA-05-4382-01" ...</span></span>
<span><span class="co">#&gt;   ..$ tmb:List of 2</span></span>
<span><span class="co">#&gt;   .. ..$ missense  :'data.frame':    502 obs. of  2 variables:</span></span>
<span><span class="co">#&gt;   .. .. ..$ sample  : chr [1:502] "TCGA-05-4244-01" "TCGA-05-4249-01" "TCGA-05-4250-01" "TCGA-05-4382-01" ...</span></span>
<span><span class="co">#&gt;   .. .. ..$ mutation: num [1:502] 163 253 270 1328 100 ...</span></span>
<span><span class="co">#&gt;   .. ..$ truncating:'data.frame':    502 obs. of  2 variables:</span></span>
<span><span class="co">#&gt;   .. .. ..$ sample  : chr [1:502] "TCGA-05-4244-01" "TCGA-05-4249-01" "TCGA-05-4250-01" "TCGA-05-4382-01" ...</span></span>
<span><span class="co">#&gt;   .. .. ..$ mutation: num [1:502] 24 45 40 206 17 18 73 31 176 108 ...</span></span>
<span><span class="co">#&gt;  $ sample.class    : Named chr [1:502] "LUAD" "LUAD" "LUAD" "LUAD" ...</span></span>
<span><span class="co">#&gt;   ..- attr(*, "names")= chr [1:502] "TCGA-05-4244-01" "TCGA-05-4249-01" "TCGA-05-4250-01" "TCGA-05-4382-01" ...</span></span>
<span><span class="co">#&gt;  $ alteration.class: Named chr [1:396] "MUT" "MUT" "MUT" "MUT" ...</span></span>
<span><span class="co">#&gt;   ..- attr(*, "names")= chr [1:396] "AKT1" "ATM" "BRAF" "CDKN2A" ...</span></span></code></pre></div>
<ul>
<li>Save the <code>run_data</code> and check the introduction vignette
to see how to run selectX to discover EDs.</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="sessioninfo">SessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Print the sessionInfo</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-conda-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: AlmaLinux 9.3 (Shamrock Pampas Cat)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS/LAPACK: /mnt/ndata/arvind/envs/selectsim_R/lib/libopenblasp-r0.3.29.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Zurich</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] tictoc_1.2.1      dplyr_1.1.4       SelectSim_0.0.1.5</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] sass_0.4.9         generics_0.1.3     tidyr_1.3.1        rstatix_0.7.2     </span></span>
<span><span class="co">#&gt;  [5] stringi_1.8.7      lattice_0.22-6     digest_0.6.37      magrittr_2.0.3    </span></span>
<span><span class="co">#&gt;  [9] evaluate_1.0.3     grid_4.4.2         iterators_1.0.14   fastmap_1.2.0     </span></span>
<span><span class="co">#&gt; [13] plyr_1.8.9         Matrix_1.7-3       foreach_1.5.2      doParallel_1.0.17 </span></span>
<span><span class="co">#&gt; [17] jsonlite_2.0.0     backports_1.5.0    Formula_1.2-5      purrr_1.0.4       </span></span>
<span><span class="co">#&gt; [21] doRNG_1.8.6.2      scales_1.3.0       codetools_0.2-20   textshaping_0.4.0 </span></span>
<span><span class="co">#&gt; [25] jquerylib_0.1.4    abind_1.4-5        cli_3.6.4          rlang_1.1.5       </span></span>
<span><span class="co">#&gt; [29] munsell_0.5.1      cachem_1.1.0       yaml_2.3.10        tools_4.4.2       </span></span>
<span><span class="co">#&gt; [33] parallel_4.4.2     reshape2_1.4.4     ggsignif_0.6.4     colorspace_2.1-1  </span></span>
<span><span class="co">#&gt; [37] ggplot2_3.5.1      ggpubr_0.6.0       rngtools_1.5.2     RcppZiggurat_0.1.8</span></span>
<span><span class="co">#&gt; [41] Rfast_2.1.0        broom_1.0.8        vctrs_0.6.5        R6_2.6.1          </span></span>
<span><span class="co">#&gt; [45] ggridges_0.5.6     lifecycle_1.0.4    stringr_1.5.1      fs_1.6.5          </span></span>
<span><span class="co">#&gt; [49] car_3.1-3          htmlwidgets_1.6.4  ragg_1.3.3         pkgconfig_2.0.3   </span></span>
<span><span class="co">#&gt; [53] desc_1.4.3         RcppParallel_5.1.9 pkgdown_2.1.1      pillar_1.10.1     </span></span>
<span><span class="co">#&gt; [57] bslib_0.9.0        gtable_0.3.6       Rcpp_1.0.14        glue_1.8.0        </span></span>
<span><span class="co">#&gt; [61] systemfonts_1.2.1  xfun_0.51          tibble_3.2.1       tidyselect_1.2.1  </span></span>
<span><span class="co">#&gt; [65] knitr_1.50         htmltools_0.5.8.1  rmarkdown_2.29     carData_3.0-5     </span></span>
<span><span class="co">#&gt; [69] compiler_4.4.2</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Arvind Iyer, Marco Mina, Miljan Petrovic, Giovanni Ciriello.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
